test/functional/p2p_seednode.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019-2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+"""
+Test seednode interaction with the AddrMan
+"""
+import time
+import random
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.messages import (CAddress, msg_addr)
+from test_framework.p2p import (
+    P2PInterface,
+    P2P_SERVICES,
+)
+
+def setup_addr_msg(num, time):
```

Comment:
In e45522cac97454eefe9f30af9f5e9df6b5bc3d65: `setup_addr_msg` seems weird. It's a for loop based on the first parameter but this function is called just once and with num=1.

Reply:
That is mostly borrowed from `p2p_addr_relay.py`, but you're right, it can be called with a bigger num so the node gets more addresses to pick up from. I'll address it.

Reply:
In fact, I realized that it has been used to fill addrman, right? So, wouldn't it be better to use `addpeeraddress`?

Reply:
Yep, that works and it's way simpler :D

Reply:
Addressed in [a178d28](https://github.com/bitcoin/bitcoin/pull/29605/commits/a178d284318f1e239fe1f2e8e8b2fd1043bed46b)